plugins/outputs/graylog/sample.conf

```diff
@@ -16,6 +16,14 @@
   ## Set to true for backward compatibility.
   # name_field_no_prefix = false
 
+  ## Connection retry options
+  ## Number of attempts to reconnect to the enpoints. If zero, only
+  ## a single connection attempt will be made. For negative values,
+  ## the plugin will try infinite number of times.
+  # reconnection_attempts = 0
+  ## Time to wait between reconnection attempts.
+  # reconnection_wait_time = "100ms"
```

Comment:
Re-trying a connection 10x a second is less than ideal. What do you think about using exponential backoff we can max out at retrying at 30 seconds. (e.g. retry every 1, 2, 4, 8, 16, 30, 30, 30, etc. seconds).

My concern with that approach is we may start to hit buffers and drop metrics, but at the same time, if we are going to be retrying a connection I think we should be "polite" and not hammer it with attempts.

Thoughts?

Reply:
Yeah let's convert this to a `max_retry_wait_time` and do an exponential back-off as you suggest. Start at 1second and go up to the specified time and clip it there...

Reply:
@powersj I just increased the default time to 30 seconds. The changing time has the drawback that we likely need to expose the start-time as an option too and I think we should keep this simple. What do you think?